data/mapping/mapping.json

```diff
@@ -72,13 +65,12 @@
           }
         },
         "doctors": {
-          "type": "text",
-          "analyzer": "gp_name_analyzer",
-          "fields": {
-            "keyword": {
-              "type": "keyword",
-              "ignore_above": 256
-            }
+          "type": "nested",
```

Comment:
I think your grep needs to be case insensitive. `jq '.[].doctors' data/input/gp-data-merged.json | grep -i 'Doctor' ` gives me matches.

ES removes full stops and commas as well as a load of non-alphanumeric characters (~,@,etc) so the gp_name_tokenizer will split 'dr. andrew smith' into ['andrew', 'smith'].

ES has a cool _analyze API to show what the ouput for a given input would be:
```
curl -XGET 'localhost:9200/profiles/_analyze?pretty' -d'
{
  "analyzer" : "gp_name_analyzer",
  "text" : "Dr. Andrew Smith"
}'
```